swift-sdk/IterableAPI.swift

```diff
@@ -151,8 +151,8 @@ import UIKit
         implementation?.setEmail(email, authToken: authToken, merge: merge, successHandler: successHandler, failureHandler: failureHandler)
     }
     
-    public static func setUserId(_ userId: String?, _ authToken: String? = nil, merge: Bool? = nil, _ successHandler: OnSuccessHandler? = nil, _ failureHandler: OnFailureHandler? = nil) {
-        implementation?.setUserId(userId, authToken: authToken, merge: merge,successHandler: successHandler, failureHandler: failureHandler)
+    public static func setUserId(_ userId: String?, _ authToken: String? = nil, merge: Bool? = nil, _ successHandler: OnSuccessHandler? = nil, _ failureHandler: OnFailureHandler? = nil, _ isAnon: Bool = false) {
+        implementation?.setUserId(userId, authToken: authToken, merge: merge,successHandler: successHandler, failureHandler: failureHandler, isAnon: isAnon)
```

Comment:
Are these both `isAnon` and `merge` required to make the anonymous flow work correctly. setUserId method might get more complicated with additional two flags for one feature. Should developers be setting the `isAnon` to true? 
whether a user is anonymous should be something only SDK should understand and process internally.

Reply:
`isAnon` is needed with the current flow since the annoymous user id is local storage is cleared if it is not provided currently. We can brainstorm a way to re-architect it correctly

Reply:
The flag is used internally. The end user will never have to set the flag. Currently it is only used internally to set an anonymous user.

Reply:
Actually it is public facing. We should have the manager call the internal iterable API.